prow/plugins/releasenote/releasenote.go

```diff
@@ -226,10 +226,10 @@ func shouldHandlePR(pr *github.PullRequestEvent) bool {
 		return false
 	}
 
-	// Ignoring unrelated PR labels prevents duplicate release note messages
+	// Ignoring unrelated PR labels prevents duplicate release note messages, merged PRs also ignored
 	if pr.Action == github.PullRequestActionLabeled {
 		for _, rnLabel := range allRNLabels {
-			if pr.Label.Name == rnLabel {
+			if pr.Label.Name == rnLabel && !pr.PullRequest.Merged {
```

Comment:
So we still fire on `Edited` though? And if *that* is desired, why bother with iterating?

Not sure about the actual logic TBH. I think it is just never right to add the actual `do-not-merge/...` label (and probably an associated comment) on a merged PR, but that's it. So personally, I'd focus on specifically not adding  that label when the PR is merged.

I don't know the expectations for the other labels, so I'd keep their behavior as it is.

Reply:
I misunderstood initially your comment but I think, probably the condition should be like:

` if pr.Action == github.PullRequestActionLabeled && !pr.PullRequest.Merged `

We still fire on edited in my opinion.

Reply:
implemented